plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -857,9 +857,9 @@ class FossId internal constructor(
         }
 
         val matchedLines = mutableMapOf<Int, MatchedLines>()
-        val snippets = runBlocking(Dispatchers.IO) {
-            pendingFiles.map {
-                async {
+        val snippets = pendingFiles.associateWith {
```

Comment:
The new implementation is now fully sequential; every concurrent processing of files has been turned off. Could this be a performance issue if there are many pending files?

Reply:
This is correct. Let's take an example with _n_ pending files and _s_ snippets in total.

Before the changes, one request was made to list the snippets per pending file, so n requests in total. Then, one request to list the matched lines _per_ snippet. Therefore _n + s_ in total, done concurrently.

With the changes, there is now _n + min(s, limit)_ requests in total, done sequentially,

I would assume this is less efficient but this is nevertheless a huge improvement for diminishing the amount of requests, as soon as the amount of snippet is growing.

Unfortunately, with the introduction of lazy loading, the initial concurrency cannot be get back. However, all the matched lines for all snippets of _one_ pending file can be made concurrently ?

Any suggestion ?

Reply:
My guts feeling is _n << s_, by a factor 10 as least.

Reply:
I put again the code to list snippets of a pending file concurrently.

As we discussed, the proper solution to avoid all these problem would be to put the listing and mapping code together.
However, this is a big refactoring and I worry that the snippet choice feature relies on having all snippets information at once for a given pending file.

Additionally, it seems FossID is moving toward OpenAPI to replace the legacy API, hence I would like to wait for their new release first, before doing such a refactoring.